### ISSUE TITLE
setProps in Server Render?

### ISSUE THREAD

**Issue by leeola** [facebook/react#1361]
## TL;DR

 `component.setProps({})` doesn't seem to work on the server. Since "mounting" isn't really a thing on the server, it appears that you can't update properties on a component instance.
## Not TL;DR

I'm not positive what the "React way" would be for this scenario, so let me try to explain.

I've got some "isomorphic" code that handles clientside actions. This code of course runs on the server as well, to render out react pages.

In the browser, my strategy for using React is to mount a root component, and update that as new data comes in from server requests, socket connections, etc. The key here, is that the client gets an _(mostly)_ empty component, and then updates it as data comes in.

The server on the otherhand has trouble doing this with the same code. Since the server never "mounts" the component, i have a hard time updating the component's properties. `comp.setProps()` returns an error because it requires the component to be mounted.. but as far as i am aware, this isn't possible on the Server.

Currently i am passing the whole render process off to a custom render function _(not attached to the React componet)_ which manually merges new data with the `component.props` object.

This allows for an isomorphic code base, but since i am manually merging objects on `component.props`, it feels dirty and brittle.

So, how would a sane person mutate a component's properties, prior to rendering, on the server?
